rfcs/text/pe/pe.yml

```
@@ -0,0 +1,257 @@
+---
+- name: pe
+  type: group
+  fields:
+    - name: icon
+      level: extended
+      type: object
+      description: >
+        Information of embedded program icon.
+      fields:
+        - name: hash
+          level: extended
+          description: >
+            Hash of the header for the PE file
+          type: object
+          fields:
+            - name: md5
+              level: extended
+              type: keyword
+              description: >
+                MD5 hash of raw icon data.
+              example: 5aa1aa0f2b4be70397a1e9e2b87627cd
+
+            - name: sha1
+              level: extended
+              type: keyword
+              description: >
+                SHA1 hash of raw icon data.
+              example: e8ff6bf77c652168f958ab4194af48b75ef1e4dd
+
+            - name: sha256
+              level: extended
+              type: keyword
+              description: >
+                SHA256 hash of raw icon data.
+              example: d81d0dbc5e691ea1082db0a8f82f978f3768d46d23049935dff2d935f932d1a2
+
+            - name: dhash
+              level: extended
+              type: keyword
+              description: >
+                Difference Hash (dhash) to find files with a visually similar icon or thumbnail.
+              example: b806e17c8e330d82
+
+    - name: debug
+      level: extended
+      type: nested
+      description: Debug information, if present
+      fields:
+        - name: offset
+          level: extended
+          type: keyword
+          description: Debug offset information.
+          example: 1296336
+
+        - name: size
+          level: extended
+          type: long
+          format: bytes
+          description: Size of the debug information.
+          example: 816
+
+        - name: type_str
+          level: extended
+          type: keyword
+          description: Format of the debug information.
+          example: IMAGE_DEBUG_TYPE_POGO
+
+        - name: type
+          level: extended
+          type: keyword
+          description: Information type generated by the debug options.
+          example: 13
+
+        - name: timestamp
+          level: extended
+          type: date
+          description: Timestamp of the debug information.
+          example: "2020-11-05T17:25:47.000Z"
+
+    - name: imports
+      level: extended
+      type: flattened
+      description: List of all imported functions
+      example: { "library_name" : "mscoree.dll", "imported_functions" : "GetFileVersionInfoSizeA" }
```

I think YAML actually will interpret this as an inline object
```suggestion
      example: '{ "library_name" : "mscoree.dll", "imported_functions" : "GetFileVersionInfoSizeA" }'
```

---

I'm not entirely sure how to document possible fields in a `flattened` type, but these should definitely be an object with queryable fields. Not sure if it makes sense to include an `example` in a `flattened` field either.

---

Yes, the quotes are necessary - good catch @rw-access!

I don't think any `flattened` example will be comprehensive by its nature, but I think it's useful to have a simple example (e.g., what's done with the [labels](https://www.elastic.co/guide/en/ecs/current/ecs-base.html#field-labels) field) showing the field's intended value is an object.